src/main/java/duke/commands/Command.java

```diff
@@ -0,0 +1,42 @@
+package duke.commands;
+
+import duke.exceptions.DukeException;
+import duke.utils.Storage;
+import duke.utils.TaskList;
+import duke.utils.Ui;
+
+/**
+ * Parent class for all Commands.
+ * Implements the logic that is fundamental to all Commands.
+ */
+public class Command {
```

Comment:
You are definitely right. I was debating over whether this is an ideal default behaviour. The reason I went down this path is because I figured every command ought to result in the file being saved as a safety mechanism. But thinking a little further, this seems wasteful, as some commands never make any modifications, so saving the file is a waste of time and computational resources. 

I'll update this in a future commit. Thanks for pointing it out :)